### PR TITLE
feat: bump up cosmos@v0.45.6

### PR DESCRIPTION
**Pull request by dudong2**
## Description

bump up cosmos-sdk v0.45.6
closes: #575 

## Checklist:

- [ ] I followed the [contributing guidelines](https://github.com/line/lbm-sdk/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/line/lbm-sdk/blob/main/CODE_OF_CONDUCT.md).
- [x] I have added a relevant changelog to `CHANGELOG.md`
- [x] I have added tests to cover my changes.
- [x] I have updated the documentation accordingly.
- [x] I have updated API documentation `client/docs/swagger-ui/swagger.yaml`

### CODE REVIEW

File: types/query/filtered_pagination.go [hunk 1]
```diff
@@ -113,3 +117,138 @@ func FilteredPaginate(
...
+
+// GenericFilteredPaginate does pagination of all the results in the PrefixStore based on the
+// provided PageRequest. `onResult` should be used to filter or transform the results.
+// `c` is a constructor function that needs to return a new instance of the type T (this is to
+// workaround some generic pitfalls in which we can't instantiate a T struct inside the function).
+// If key is provided, the pagination uses the optimized querying.
+// If offset is used, the pagination uses lazy filtering i.e., searches through all the records.
+// The resulting slice (of type F) can be of a different type than the one being iterated through
+// (type T), so it's possible to do any necessary transformation inside the onResult function.
+// TODO(dudong2): after upgrade go 1.18, remove below comments for GenericFilteredPaginate()
```
Comment (zemyblue) [inline types/query/filtered_pagination.go:129]: Why can't we update golang version?

File: x/authz/client/cli/query.go [hunk 2]
```diff
@@ -91,3 +93,101 @@ $ %s query %s grants link1skjw.. link1skjwj.. %s
 	flags.AddPaginationFlagsToCmd(cmd, "authorizations")
```
Comment (zemyblue) [inline x/authz/client/cli/query.go:93]: Please check why the query string is different.

cosmos-sdk, "grants"
https://github.com/cosmos/cosmos-sdk/blob/f2d94445c0f5f52cf5ed999b81048b575de94964/x/authz/client/cli/query.go#L93

But, lbm-sdk, "authorizations"

File: x/authz/keeper/grpc_query.go [hunk 3]
```diff
@@ -90,8 +89,276 @@ func (k Keeper) Grants(c context.Context, req *authz.QueryGrantsRequest) (*authz
...
+	}, nil
+}
+
 // unmarshal an authorization from a store value
 func unmarshalAuthorization(cdc codec.BinaryCodec, value []byte) (v authz.Grant, err error) {
 	err = cdc.Unmarshal(value, &v)
 	return v, err
 }
+
+// TODO(dudong2): after upgrade go 1.18, remove below comments to use GenericFilteredPaginate()
```
Comment (zemyblue) [inline x/authz/keeper/grpc_query.go:203]: please, check it also.

**Review by zemyblue — COMMENTED** [review]
Please check the changes of `swagger.yam` and `static.go` files

File: x/authz/keeper/grpc_query.go (hunk 3 above)
Comment (dudong2) [reply x/authz/keeper/grpc_query.go]: fixed it

File: .github/workflows/sims.yml [hunk 4]
```diff
@@ -25,7 +25,7 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/setup-go@v3.2.1
         with:
-          go-version: 1.15
+          go-version: 1.16
```
[the remark this answers is not in the document]
Comment (dudong2) [reply .github/workflows/sims.yml]: upgraded to 1.18

**Comment by zemyblue** [comment]
Please, update README and related document and testing about changing golang 1.18.

File: contrib/images/simd-env/Dockerfile [hunk 5]
```diff
@@ -1,8 +1,8 @@
-FROM golang:1.17-alpine AS build
+FROM golang:1.16-alpine AS build
```
[the remark this answers is not in the document]
Comment (dudong2) [reply contrib/images/simd-env/Dockerfile]: fixed it

File: README.md [hunk 6]
```diff
@@ -1,73 +1,73 @@
...
-You can refer the sample tx commands at [here](docs/sample-tx.md) 
-Test different commands to get a broader understanding of lbm
-
+The LBM SDK is a framework for building blockchain applications in Golang.
+It is being used to build [`LBM`](https://github.com/line/lbm), the first implementation of the LINE Blockchain Mainnet.
+This is forked from [`cosmos-sdk`](https://github.com/cosmos/cosmos-sdk) at 2021-03-15.
+
+**WARNING**: Breaking changes may occur because this repository is still in the pre-release development phase.
+
+**Note**: Requires [Go 1.15+](https://golang.org/dl/)
```
[the remark this answers is not in the document]
Comment (dudong2) [reply README.md:7]: fixed it

File: types/query/filtered_pagination.go (hunk 1 above)
Comment (dudong2) [reply types/query/filtered_pagination.go]: updated it

File: x/authz/client/cli/query.go (hunk 2 above)
Comment (dudong2) [reply x/authz/client/cli/query.go]: changed it. thx!

**Review by zemyblue — APPROVED** [review]
Please check again swagger's changes.

File: baseapp/baseapp_test.go [hunk 7]
```diff
@@ -2043,3 +2043,29 @@ func TestBaseApp_EndBlock(t *testing.T) {
...
+}
+
+func TestSnapshotManager(t *testing.T) {
+	app := newBaseApp(t.Name())
+	require.Nil(t, app.SnapshotManager())
+
+	tempDir := t.TempDir()
+	snapshotDB, err := sdk.NewLevelDB("metadata", tempDir)
+	if err != nil {
+		panic(err)
```
Comment (0Tech) [inline baseapp/baseapp_test.go:2063]: Isn't `require.NoError()` more appropriate here?

```suggestion
		require.NoError(t, err)
```

File: baseapp/baseapp_test.go [hunk 8]
```diff
@@ -2043,3 +2043,29 @@ func TestBaseApp_EndBlock(t *testing.T) {
...
+	}
+	snapshotStore, err := snapshots.NewStore(snapshotDB, tempDir)
+	if err != nil {
+		panic(err)
```
Comment (0Tech) [inline baseapp/baseapp_test.go:2067]: Same as above.
```suggestion
		require.NoError(t, err)
```

**Review by 0Tech — APPROVED** [review]
LGTM, with some minor comments.

File: baseapp/baseapp_test.go (hunk 7 above)
Comment (dudong2) [reply baseapp/baseapp_test.go]: changed it. thanks!

File: baseapp/baseapp_test.go (hunk 8 above)
Comment (dudong2) [reply baseapp/baseapp_test.go]: changed it. thanks!

**Pull request merged**